dts/arm/raspberrypi/rpi_pico/rp2350.dtsi

```diff
@@ -0,0 +1,402 @@
+/*
+ * Copyright (c) 2024 Andrew Featherstone
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <arm/armv8-m.dtsi>
+#include <zephyr/dt-bindings/adc/adc.h>
+#include <zephyr/dt-bindings/gpio/gpio.h>
+#include <zephyr/dt-bindings/clock/rpi_pico_rp2350_clock.h>
+#include <zephyr/dt-bindings/i2c/i2c.h>
+#include <zephyr/dt-bindings/regulator/rpi_pico.h>
+#include <zephyr/dt-bindings/reset/rp2350_reset.h>
+#include <mem.h>
+
+#ifndef RPI_PICO_DEFAULT_IRQ_PRIORITY
+#define RPI_PICO_DEFAULT_IRQ_PRIORITY 7
+#endif
+
+/ {
+	aliases {
+		die-temp0 = &die_temp;
+	};
+
+	cpus {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		cpu0: cpu@0 {
+			compatible = "arm,cortex-m33";
+			reg = <0>;
+		};
+
+		cpu1: cpu@1 {
+			compatible = "arm,cortex-m33";
+			reg = <1>;
+		};
+	};
+
+	clocks {
```

Comment:
at first sight it seems like there's a significant overlap with this and the current rp2040.dtsi file, would it make sense to have a "common" file and extend the specific ones? saves duplicated works if you end up updating drivers that need a matching change in the dtsi

Reply:
Some cases under the dts folder have created "...common.dtsi" as a common ancestor.
The rp2040 and rp2350 have different structures because their cores are different, and they can't have a common ancestor.
So, I think it is good to leave them separated rather than crop and separate identical parts of these configurations into common.dtsi.

For now, I have requested to proceed with this policy.

Reply:
cool, thanks for clarifying